mewarpx/mewarpx/diags_store/checkpoint_diagnostic.py

```diff
@@ -13,32 +15,91 @@
 
 class CheckPointDiagnostic(WarpXDiagnostic):
     def __init__(self, diag_steps,
-                 name=init_restart_util.default_checkpoint_name, **kwargs):
+                 name=init_restart_util.default_checkpoint_name,
+                 clear_old_checkpoints=True, **kwargs):
         """
         This class is a wrapper for creating checkpoints from which
-        to restart a simulation from
+        to restart a simulation from. Adding flux diagnostic data to
+        checkpoints are supported, but since this class has to be initialized
+        before the simulation and flux diagnostics are initialized after
+        the simulation, the user is responsible for adding the ``flux_diag``
+        attribute to this object in a simulation input file.
 
         Arguments:
             diag_steps (int): Run the diagnostic with this period.
                 Also plot on this period if enabled.
             name (str): The name of the diagnostic to be passed into the
                 picmi checkpoint diagnostic.
+            clear_old_checkpoints (bool): If True old checkpoints will be
+                deleted after new ones are created.
             kwargs: For a list of valid keyword arguments see
                 :class:`mewarpx.diags_store.diag_base.WarpXDiagnostic`
         """
-        self.diag_steps = diag_steps
+        self.checkpoint_steps = diag_steps
         self.name = name
+        self.clear_old_checkpoints = clear_old_checkpoints
+        self.flux_diag = None
 
-        super(CheckPointDiagnostic, self).__init__(diag_steps, **kwargs)
+        super(CheckPointDiagnostic, self).__init__(diag_steps+1, **kwargs)
 
         self.write_dir = self.DIAG_DIR
         self.add_checkpoint()
 
+        # note that WarpX diagnostics (including checkpoints) are output
+        # after the "afterstep" functions, therefore we install this function
+        # before steps and add 1 to the diag_steps above
+        callbacks.installbeforestep(self.checkpoint_manager)
+
     def add_checkpoint(self):
         diagnostic = picmi.Checkpoint(
-            period=self.diag_steps,
+            period=self.checkpoint_steps,
             name=self.name,
             write_dir=self.write_dir
         )
 
         mwxrun.simulation.add_diagnostic(diagnostic)
+
+    def checkpoint_manager(self):
+        """Function executed on checkpoint steps to perform various tasks
+        related to checkpoint management. These include copying the flux
+        diagnostic data needed for a restart as well as deleting old
+        checkpoints.
+        """
+        # only the root processor should execute this function
+        if not self.check_timestep() or mwxrun.me != 0:
+            return
+
+        # Copy flux diagnostics, if present, to load when restarting.
+        if (self.flux_diag is not None
+            and self.flux_diag.last_run_step == mwxrun.get_it() - 1
```

Comment:
I think if you have MAX_STEPs in the unit test go to 9 rather than 8 you should be able to check for the restart file existing at step 8, if my glance at that file was correct.

Reply:
I'll mark this as resolved since the fix you pointed out above will take care of this as well.